### PR TITLE
Refactor IP Tool/Helper

### PR DESCRIPTION
**Pull request by nibanks**
Refactors the IP tool/helper API a bit. Allows for stubbing out logging. Allows for using private registration and function table.

### CODE REVIEW

File: src/tools/ip/quicip.h [hunk 1]
```diff
@@ -17,15 +17,28 @@
 #pragma once
 
 #include <msquic.h>
+#include <msquicp.h>
```
Comment (ThadHouse) [inline src/tools/ip/quicip.h:20]: What is this using out of msquicp? (Other then the self signed security thing).

Comment (nibanks) [reply src/tools/ip/quicip.h:20]: That's it.

File: src/tools/ip/quicip.h [hunk 2]
```diff
@@ -72,8 +98,20 @@ ClientConnectionCallback(
...
     switch (Event->Type) {
+    case QUIC_CONNECTION_EVENT_SHUTDOWN_INITIATED_BY_TRANSPORT:
+        if (!Context->Success && Context->Status == QUIC_IP_DEFAULT_STATUS) {
+            QUIC_PRINTF("Connection Shutdown, 0x%x!\n", Event->SHUTDOWN_INITIATED_BY_TRANSPORT.Status);
+            Context->Status = Event->SHUTDOWN_INITIATED_BY_TRANSPORT.Status;
+        }
+        break;
+    case QUIC_CONNECTION_EVENT_SHUTDOWN_INITIATED_BY_PEER:
+        if (!Context->Success && Context->Status == QUIC_IP_DEFAULT_STATUS) {
+            QUIC_PRINTF("Connection Shutdown by Peer, 0x%llx!\n", Event->SHUTDOWN_INITIATED_BY_PEER.ErrorCode);
```
Comment (nibanks) [inline src/tools/ip/quicip.h:111]: Apparently, Linux doesn't like this:
```
/home/vsts/work/1/s/src/tools/ip/quicip.h:24:56: error: format ‘%llx’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘QUIC_UINT62 {aka long unsigned int}’ [-Werror=format=]
 #define QUIC_PRINTF(fmt, ...) printf(fmt, ##__VA_ARGS__)
                                                        ^
/home/vsts/work/1/s/src/tools/ip/quicip.h:73:64:
             QUIC_PRINTF("Stream Peer Send Aborted, 0x%llx!\n", Event->PEER_SEND_ABORTED.ErrorCode);
```
What's the best way forward?

Comment (ThadHouse) [reply src/tools/ip/quicip.h:111]: #include <inttypes.h>

Then instead of `0x%llx!` use `%"PRIx64"` (The quotes are important here)

Comment (ThadHouse) [reply src/tools/ip/quicip.h:111]: Just noticed because C++ you need spaces between the end quotes and the macro

Comment (nibanks) [reply src/tools/ip/quicip.h]: Just decided to remove them all together.

**Pull request merged**